korge/src/commonMain/kotlin/com/soywiz/korge/view/Sprite.kt

```diff
@@ -235,11 +243,17 @@ open class Sprite(
         animationType = type
         animationRequested = true
         currentAnimation?.let {
-            this.animationNumberOfFramesRequested = if (!looped) animationCyclesRequested * it.spriteStackSize else it.spriteStackSize
+            this.animationNumberOfFramesRequested = when {
```

Comment:
That's what came out of my mind. I feel, this could be handled more elegantly though. Maybe you have ideas